android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java

```diff
@@ -248,4 +261,15 @@ private AccountInfo getThisAccountInfo() {
         accountInfo.isImported = mIsImported;
         return accountInfo;
     }
+
+    @Override
+    public void onTransactionStatusChanged(TransactionInfo txInfo) {
+        mWalletTxCoinAdapter.onTransactionUpdate(txInfo);
+    }
+
+    private void initState() {
+        mWalletTxCoinAdapter =
+                new WalletCoinAdapter(WalletCoinAdapter.AdapterType.VISIBLE_ASSETS_LIST);
+        pendingTxHelper = new PendingTxHelper(getEthTxService(), accountInfos, false, null);
```

Comment:
You pass null object as `accountInfos` here as `initState` is called before `accountInfos` is initialized inside `fetchAccountInfo`

Reply:
The whole `initState` has to be called after `accountInfos = keyringInfo.accountInfos;`. But as I mentioned above I don't understand why do we create `PendingTxHelper` here as we never call it's method `fetchTransactions` to fetch transactions and after that we are trying to modify it.

Reply:
[Update] Removed the unused PendingTxHelper usage and refactoring.